uarray/_uarray_dispatch.cxx

```diff
@@ -353,6 +353,80 @@ struct SetBackendContext
     }
 };
 
+struct Dispatchable
+{
+  PyObject_HEAD
+
+  py_ref value;
+  py_ref type;
+  bool coercible;
+
+  static void dealloc(Dispatchable * self)
+  {
+    self->~Dispatchable();
+    Py_TYPE(self)->tp_free(self);
+  }
+
+  static PyObject * get_value(Dispatchable * self, void *)
+  {
+    return py_ref::ref(self->value).get();
+  }
+
+  static PyObject * get_type(Dispatchable * self, void *)
+  {
+    return py_ref::ref(self->type).get();
+  }
+
+  static PyObject * get_coercible(Dispatchable * self, void *)
+  {
+    return PyBool_FromLong(self->coercible);
+  }
+
+  static PyObject * new_(PyTypeObject * type, PyObject * args, PyObject * kwargs)
+  {
+    auto self = reinterpret_cast<Dispatchable *>(type->tp_alloc(type, 0));
+    if (self == nullptr)
+      return nullptr;
+
+    // Placement new
+    self = new (self) Dispatchable;
+    return reinterpret_cast<PyObject *>(self);
+  }
+
+  static int init(
+    Dispatchable * self, PyObject * args, PyObject * kwargs)
+  {
+    static const char *kwlist[] = {"value", "type", "coercible", nullptr};
+    PyObject * value, * type;
+    int coercible = 1;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwargs,
+                                      "OO|p", (char**)kwlist,
+                                      &value, &type, &coercible))
+      return -1;
+
+    self->value = py_ref::ref(value);
+    self->type = py_ref::ref(type);
+    self->coercible = coercible;
+    return 0;
+  }
+
+  static int traverse(Dispatchable * self, visitproc visit, void * arg)
+  {
+    Py_VISIT(self->value);
+    Py_VISIT(self->type);
+    return 0;
+  }
+
+  static PyObject * repr(Dispatchable * self)
+  {
+    return PyUnicode_FromFormat("<uarray Dispatchable, value=%s, type=%s, coercible=%s>",
+      PyObject_Repr(self->value.get()),
+      PyObject_Repr(self->type.get()),
+      PyObject_Repr(py_ref::ref(PyBool_FromLong(self->coercible)).get())
```

Comment:
https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_FromFormat

Use `%R` to format the repr from a `PyObject *`.